cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -161,16 +168,38 @@ public CruiseControlApiImpl(Vertx vertx) {
                 .get(port, host, path, response -> {
                     response.exceptionHandler(result::fail);
                     if (response.statusCode() == 200 || response.statusCode() == 201) {
-                        String userTaskID = response.getHeader(USER_ID_HEADER);
+                        String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
                         response.bodyHandler(buffer -> {
-                            JsonObject json = buffer.toJsonObject();
+                            JsonObject jsonUserTask = buffer.toJsonObject().getJsonArray("userTasks").getJsonObject(0);
+                            JsonObject json = new JsonObject()
+                                    .put("Status", jsonUserTask.getString("Status"))
+                                    .put("summary", ((JsonObject) Json.decodeValue(jsonUserTask.getString("originalResponse"))).getJsonObject("summary"));
+
                             if (json.containsKey(CC_REST_API_ERROR_KEY)) {
                                 result.fail(json.getString(CC_REST_API_ERROR_KEY));
                             } else {
                                 CruiseControlResponse ccResponse = new CruiseControlResponse(userTaskID, json);
                                 result.complete(ccResponse);
                             }
                         });
+                    } else if (response.statusCode() == 500) {
+                        response.bodyHandler(buffer -> {
+                            String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
+                            JsonObject json = buffer.toJsonObject();
+                            CruiseControlResponse ccResponse = new CruiseControlResponse(userTaskID, json);
+                            if (json.containsKey(CC_REST_API_ERROR_KEY)) {
+                                // in this case "NotEnoughValidWindowsException" is the cause of the main exception
+                                // and it's provided by the REST API just in the "stackTrace"
+                                if (json.getString(CC_REST_API_STACKTRACE_KEY).contains("NotEnoughValidWindowsException")) {
```

Comment:
I don't think `NotEnoughWindows` would apply to the user tasks endpoint? It will just return a list of the past endpoint requests. This error may be present in the `originalResponse` of one of the tasks listed there but will not be in the error returned by the end point. So I don't think we need this if branch.

Reply:
Maybe I saw this `NotEnoughWindows` exception when I was using the userTasts during the rebalance proposal pending. Now we moved to poll using the rebalance endpoint and not the userTasks anymore, so agree I will remove this branch.